src/components/share-latex/share-latex.tsx

```diff
@@ -165,19 +166,21 @@ export default function ShareLatex(props: {
     window.open(fileUrl)
   }
 
-  let engine: any;
+  const [texEngine, setTexEngine] = createSignal<PdfTeXEngine>()
   const onCompile = async () => {
+    let engine = texEngine()
     if (!engine) {
-      engine = new (globalThis as any).PdfTeXEngine()
+      engine = new (globalThis as unknown as { PdfTeXEngine: { new(): PdfTeXEngine } }).PdfTeXEngine()
```

Comment:
Ideally the clean way to do this is to augment the `global` module with the class type (say with `globals.d.ts`). But I couldn't get this to work with solidjs on first attempt.